lib/trino-record-decoder/src/test/java/io/trino/decoder/avro/TestAvroDecoder.java

```diff
@@ -364,10 +363,7 @@ public void testSchemaEvolutionToIncompatibleType()
                 .toString();
 
         assertThatThrownBy(() -> decodeRow(originalIntData, ImmutableSet.of(stringColumnReadingIntData), ImmutableMap.of(DATA_SCHEMA, changedTypeSchema)))
-                .isInstanceOf(TrinoException.class)
-                .hasCauseExactlyInstanceOf(AvroTypeException.class)
-                .hasStackTraceContaining("Found int, expecting string")
-                .hasMessageMatching("Decoding Avro record failed.");
+                .isInstanceOf(AssertionError.class);
```

Comment:
while this works the test should probably be modified to assert that `decodeRow` returns `Optional.empty` instead.

Reply:
Maybe here : 
```
    private static Map<DecoderColumnHandle, FieldValueProvider> decodeRow(byte[] avroData, Set<DecoderColumnHandle> columns, Map<String, String> dataParams)
    {
        RowDecoder rowDecoder = DECODER_FACTORY.create(dataParams, columns);
        return rowDecoder.decodeRow(avroData)
                .orElseThrow(AssertionError::new);
    }
```
Instead of assertion arrow - some IllegalStateException("Problems during Decode phase") ?

Reply:
Yeah, a proper message to match on would be useful.